cardano-node/src/Cardano/CLI/Genesis.hs

```diff
@@ -21,17 +32,102 @@ import           System.Posix.Files (ownerReadMode, setFileMode)
 import           System.Directory (emptyPermissions, readable, setPermissions)
 #endif
 
+import qualified Cardano.Chain.Common as CC.Common
 import           Cardano.Chain.Delegation hiding (Map, epoch)
 import           Cardano.Chain.Genesis
 import           Cardano.Crypto (SigningKey (..))
-import qualified Cardano.Crypto.Signing as Crypto
+import qualified Cardano.Crypto as Crypto
 
 import           Cardano.CLI.Key
 import           Cardano.CLI.Ops
 
 
-dumpGenesis :: CLIOps IO -> FilePath -> GenesisData -> GeneratedSecrets -> IO ()
-dumpGenesis CLIOps{..} outDir genesisData GeneratedSecrets{..} = do
+newtype NewDirectory =
+  NewDirectory FilePath
+  deriving (Eq, Ord, Show, IsString)
+
+newtype GenesisFile =
+  GenesisFile FilePath
+  deriving (Eq, Ord, Show, IsString)
+
+-- | Parameters required for generation of new genesis.
+data GenesisParameters where
```

Comment:
Does that need to be a `GADT` style `data` decl?

Reply:
Done.